test/visual/mpl/circuit/test_circuit_matplotlib_drawer.py

```diff
@@ -64,19 +68,87 @@ def cwd(path):
         os.chdir(oldpwd)
 
 
-class TestMatplotlibDrawer(QiskitTestCase):
+class TestCircuitMatplotlibDrawer(QiskitTestCase):
     """Circuit MPL visualization"""
 
     def setUp(self):
         super().setUp()
-        self.circuit_drawer = TestMatplotlibDrawer.save_data_wrap(
+        self.circuit_drawer = TestCircuitMatplotlibDrawer.save_data_wrap(
             _matplotlib_circuit_drawer, str(self)
         )
 
+        if not os.path.exists(FAILURE_DIFF_DIR):
+            os.makedirs(FAILURE_DIFF_DIR)
+
+        if not os.path.exists(RESULTDIR):
+            os.makedirs(RESULTDIR)
+
     def tearDown(self):
         super().tearDown()
         mpl_close("all")
 
+    @staticmethod
+    def _image_path(image_name):
+        return os.path.join(RESULTDIR, image_name)
+
+    @staticmethod
+    def _reference_path(image_name):
+        return os.path.join(TEST_REFERENCE_PATH, image_name)
+
+    @staticmethod
+    def _new_gray(size, color):
+        img = Image.new("L", size)
+        drawing = ImageDraw.Draw(img)
+        drawing.rectangle((0, 0) + size, color)
+        return img
+
+    @staticmethod
+    def _black_or_b(diff_image, image, reference, opacity=0.85):
+        """Copied from https://stackoverflow.com/a/30307875"""
```

Comment:
I don't know if this is pre-existing, but I don't think we're satisfying the licensing conditions of StackOverflow's Creative Commons Attribution+ShareAlike if we're copying code.  If you didn't newly introduce this we don't need to fix it immediately in this PR, but we might need to look at it.

Reply:
This is pre-existing, letl me know how you would like this addressed.  We can always make that a separate issue/PR.